--- git status ---
On branch f1tenth-environment
Your branch is up to date with 'origin/f1tenth-environment'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
  (commit or discard the untracked or modified content in submodules)
	modified:   f1tenth_assets (untracked content)
	modified:   source/extensions/omni.isaac.orbit/omni/isaac/orbit/sensors/lidar/lidar.py
	modified:   source/extensions/omni.isaac.orbit_tasks/omni/isaac/orbit_tasks/f1tenth/agents/rsl_rl_ppo_cfg.py

no changes added to commit (use "git add" and/or "git commit -a") 


--- git diff ---
diff --git a/source/extensions/omni.isaac.orbit/omni/isaac/orbit/sensors/lidar/lidar.py b/source/extensions/omni.isaac.orbit/omni/isaac/orbit/sensors/lidar/lidar.py
index 1060a84..241dae6 100644
--- a/source/extensions/omni.isaac.orbit/omni/isaac/orbit/sensors/lidar/lidar.py
+++ b/source/extensions/omni.isaac.orbit/omni/isaac/orbit/sensors/lidar/lidar.py
@@ -17,9 +17,9 @@ import omni.kit.commands
 import omni.usd
 from omni.isaac.core.prims import XFormPrimView
 
-from omni.isaac.core.utils.extensions import enable_extension
+# from omni.isaac.core.utils.extensions import enable_extension
 
-enable_extension("omni.isaac.range_sensor")  # required by OIGE
+# enable_extension("omni.isaac.range_sensor")  # required by OIGE
 # enable_extension("omni.isaac.RangeSensorSchema")  # required by OIGE
 from omni.isaac.range_sensor import _range_sensor
 import omni.isaac.RangeSensorSchema as RangeSensorSchema
diff --git a/source/extensions/omni.isaac.orbit_tasks/omni/isaac/orbit_tasks/f1tenth/agents/rsl_rl_ppo_cfg.py b/source/extensions/omni.isaac.orbit_tasks/omni/isaac/orbit_tasks/f1tenth/agents/rsl_rl_ppo_cfg.py
index a82b7b2..4875c4d 100644
--- a/source/extensions/omni.isaac.orbit_tasks/omni/isaac/orbit_tasks/f1tenth/agents/rsl_rl_ppo_cfg.py
+++ b/source/extensions/omni.isaac.orbit_tasks/omni/isaac/orbit_tasks/f1tenth/agents/rsl_rl_ppo_cfg.py
@@ -24,6 +24,7 @@ class F1tenthPPORunnerCfg(RslRlOnPolicyRunnerCfg):
     save_interval = 10
     experiment_name = "f1tenth"
     empirical_normalization = False
+    device = "cuda:0"
     policy = RslRlPpoActorCriticCfg(
         init_noise_std=1.0,
         actor_hidden_dims=[512, 256, 128],